proposed/2022/vulnerabilities-in-restore.md

```diff
@@ -80,6 +80,14 @@ This feature will be opt-in to start and gather feedback from developers.
 
 To enable the feature, a developer can add `<NuGetAudit>enable</NuGetAudit>` to their project file as a MSBuild property. To disable the feature, a developer can add `<NuGetAudit>disable</NuGetAudit>` or remove the property from the project file.
 
+#### Setting Vulnerability Auditing Modes
+
+There will be different modes to audit vulnerabilities based on the developer's or developer's team preference. To do this, a developer will opt-in to a feature called `<NuGetAuditMode>` which will have different modes such as `direct`, `transitive`, and `all`.
+
+These modes should be pretty straight-forward. `direct` will scan for any top-level vulnerabilities, `transitive` will scan for any transitive-level vulnerabilities, and `all` will scan for both top-level and transitive-level vulnerabilities. The default will be `direct` until the experience is ready to be `all` given that transitive vulnerabilities are the majority of vulnerability notices (90%+).
+
+When a known vulnerability is found that is of the `transitive` level, it will include the path to the project containing the top-level package and including the name and version of the package the vulnerable transitive dependency is coming from. Transitive level known vulnerabilities should not be a warning, but rather a message/informational MSBuild severity as they should not break builds but still be brought up in the Error List as informational.

```

Comment:
What do you think about moving this to the "future possibilities" section? While I definitely want to have package graph path displayed to make it easier, to me it raises questions about if adding NuGetAuditMode without showing the package graph path is acceptable (feature not implemented to spec), and whether showing the path in the audit warning messages is higher priority than other similar features, like `dotnet nuget why`.

Reply:
I tend to agree. However, there should be an easy way for developers to know the full paths (could be multiple?) so that its clear what brings the transitive in. IMO, whether disjoint feature or not, should be part of the vulnerability epic for .NET 8.

Reply:
I have been having various conversations and I think we should try to provide as much provenance in the warning messages as we can so it helps empower the developer to remediation until we can move towards our fixing experiences. `dotnet nuget why` would be a helpful command for further provenance, but we should provide enough in these warnings to give people somewhere to look initially. We have until August for much of this work.